### PR TITLE
ISPN-13951 Use the new indexing annotations

### PR DESCRIPTION
**Pull request by karesti**
https://issues.redhat.com/browse/ISPN-13951

### CODE REVIEW

**Review by fax4ever — APPROVED** [review]
LGTM +1 to merge it

File: infinispan-remote/query/src/main/java/org/infinispan/tutorial/simple/remote/query/Person.java [hunk 1]
```diff
@@ -1,24 +1,32 @@
...
    @ProtoField(number = 1)
+   @Keyword(projectable = true, sortable = true, normalizer = "lowercase", indexNullAs = "unnamed", norms = false)
    String firstName;
 
    @ProtoField(number = 2)
+   @Keyword(projectable = true, sortable = true, normalizer = "lowercase", indexNullAs = "unnamed", norms = false)
    String lastName;
 
    @ProtoField(number = 3, defaultValue = "-1")
+   @Decimal
```
Comment (fax4ever) [inline infinispan-remote/query/src/main/java/org/infinispan/tutorial/simple/remote/query/Person.java:25]: Just noticed @karesti.
Usually the decimal should be used if a decimal scale different from 0 is present, for instance 2 for a decimal value of a currency field.
In this case you can simply use the `@Basic` annotation.
Same for the string fields that do not require a normalizer or analyzer, you can use `@Basic`.

Comment (fax4ever) [reply infinispan-remote/query/src/main/java/org/infinispan/tutorial/simple/remote/query/Person.java:25]: basic means take the field as it is, without no transformation

Comment (fax4ever) [reply infinispan-remote/query/src/main/java/org/infinispan/tutorial/simple/remote/query/Person.java:25]: On a numeric field, to use a `@Decimal`with no scale  is equivalent to use a `@Basic`.
As on a string field, to use a `@Keyword` with no normalizer is equivalent to a `@Text` with no analyzer that is equivalent to using a `@Basic`.
In general I would advise to use `@Basic`, if it is possible.

**Comment by karesti** [comment]
@fax4ever I added the change

**Comment by fax4ever** [comment]
Merged, thanks!

**Pull request merged**